### ISSUE TITLE
DIY Tool is not working my Sonoff Basic R3

### ISSUE THREAD

**Issue by iosamu87** [itead/Sonoff_Devices_DIY_Tools#25]
Hi,

I updated the sonoff to version 3.3.0
I create the dedicated wireless network with computer and sonoff (which I see correctly).
I open the tool_01DIY85 but I don't see the sonoff.

what am I doing wrong ?

Thanks.

**Comment by alexbk66** [comment]
@iosamu87  Please search existing issues - there's a lot of discussion around this issue.  And if possible please delete this duplicate to keep the forum clean.

**Comment by seanchs** [comment]
@alexbk66 I have the similar issue - the tool can't find my two Basic R3 devices. Both Basic R3 has 3.3.0 firmware and connected to sonoffDiy network. MY laptop also connected to the sonoffDiy. I also disabled the Ethernet card on my laptop - only leave wireless card enabled. But the tool won't be able to show any Basic R3 device. Would you mind sharing how you finally fixed it? Thanks a lot

**Comment by iosamu87** [comment]
look, i gave up.
I attached my ttl usb and reprogrammed with testoma and mqtt broker to interface it with domoticz.

**Comment by alexbk66** [comment]
What a pain

**Comment by iosamu87** [comment]
back to the old at least it works :)

**Comment by alexbk66** [comment]
With this quality Sonoff will get at most 10-20 really enthusiastic customers who are willing to waste their time trying to make it work...

**Comment by ZZLinvec** [comment]
@seanchs Is the jumper cap of the equipment connected

**Comment by seanchs** [comment]
@ZZLinvec if you refer to the jumper that enables the DIY mode of the device, then it's YES - I did put the jumper on. Still have no luck to have the DIY tool recognize the Sonoff Basic R3. : ( Not sure what else I can do. Any guidance would be highly appreciated! Thanks,

**Comment by alexbk66** [comment]
@seanchs if you read what others say - "with some magic now it works". I had same experience.
So all you need is just some magic.

**Comment by dastrix80** [comment]
Well that's kinda stupid. Magic? As in , nothing was changed and it just worked?

**Comment by alexbk66** [comment]
> Magic? As in , nothing was changed and it just worked?
> 
Probably after various attempts it starts to work, but I was too frustrated to recall what exactly made it to work.
> im going to ask for my money back because DIY mode does NOT work
I did ask for refund - as we are doing the testing and shouldn't pay for it

**Comment by Franklin67** [comment]
i can see my Basic R3 in the DIY tool but when I try to flash it with the latest TASMOTA I always get the following error message:

"Firmware file cannot be greater than 508k; The Firmware must be Dout"

Why is it not possible to flash the Basic R3? Any ideas?

**Comment by j-keck** [comment]
@Franklin67 have you use the [tasmota-wifiman.bin](http://thehackbox.org/tasmota/tasmota-wifiman.bin) binary?

check the tasmota wiki https://github.com/arendst/Tasmota/wiki/Sonoff-DIY for the manual.

**Comment by Franklin67** [comment]
@j-keck: ja, habe ich jetzt bei 2 meiner Basic R3 gemacht und beide geflascht. Leider sind jetzt beide irgendwie tot und ich kann nichts mehr damit machen. :-(
Wenn ich die an Strom anschliesse geht kein Lämpchen mehr an und die Teile scheinen jetzt im Nirwana zu sein.
Der Flash Vorgang ist ohne Fehler durchgelaufen bis zu der Meldung, dass der Flash-Vorgang fertig ist und das Gerät nun neu startet. Das war es dann.

**Comment by j-keck** [comment]
@Franklin67 Hatte ich auch einmal bei einem Sonoff Mini.

Zur Not kannst du aber den Seriellen Ausgang anlöten und mit [esptool](https://github.com/espressif/esptool) neu flashen.

Falls das flashen nicht funktioniert, vorher ein `esptool erase_flash`.
Das hat bei meinen esp's immer funktioniert.

**Comment by alexbk66** [comment]
> @Franklin67 Hatte ich auch einmal bei einem Sonoff Mini.
> 
> Zur Not kannst du aber den Seriellen Ausgang anlöten und mit [esptool](https://github.com/espressif/esptool) neu flashen.
> 
> Falls das flashen nicht funktioniert, vorher ein `esptool erase_flash`.
> Das hat bei meinen esp's immer funktioniert.

Would be nice in English

**Comment by AgentFire** [comment]
> @Franklin67 have you use the [tasmota-wifiman.bin](http://thehackbox.org/tasmota/tasmota-wifiman.bin) binary?
> 
> check the tasmota wiki https://github.com/arendst/Tasmota/wiki/Sonoff-DIY for the manual.

There is no wifiman binary.  
There is no wiki on your link.

The tasmota (https://github.com/arendst/) project has no firmware files that can be uploaded via this tool - your tool throws weird "508k" exception on every one of them.
